code/datums/traits/negative_quirk.dm

```diff
@@ -426,8 +426,11 @@
 	var/slot_string = "limb"
 
 /datum/quirk/prosthetic_limb/on_spawn()
-	var/limb_slot = pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	var/limb_slot = pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG) // default to random
 	var/mob/living/carbon/human/H = quirk_target
+	var/client/C = GLOB.directory[ckey(quirk_holder.key)] // this was the only way I could find to get the client during on_spawn()
+	if(C)
+		limb_slot = C.prefs.read_character_preference(/datum/preference/choiced/quirk_prosthetic_limb_location)
```

Comment:
`prefs` can be null, check `if(C?.prefs)` instead

Reply:
added